src/main/java/chess/controller/WebChessController.java

```diff
@@ -0,0 +1,123 @@
+import chess.domain.position.Position;
+import chess.dto.ScoreDto;
+import chess.dto.TurnDto;
+import chess.dto.UnitsDto;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import spark.ModelAndView;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+import spark.template.handlebars.HandlebarsTemplateEngine;
+
+public class WebChessController {
+	private final Gson gson;
+	private final MoveDao moveDao;
+	private final GamesDao gamesDao;
+
+	public WebChessController(MoveDao moveDao, GamesDao gamesDao) {
+		this.gson = new GsonBuilder().create();
+		this.moveDao = moveDao;
+		this.gamesDao = gamesDao;
+	}
+
+	public void route() {
+		Spark.staticFileLocation("/templates");
+		get("/", this::index);
+		get("/init", this::init);
+		get("/new_game/:id", this::createNewGame);
+		get("/games", this::games);
+		get("/turn/:id", this::turn);
+		get("/board/:id", this::board);
+		get("/score/:id", this::score);
+		post("/move/:id", this::move);
+		post("/users", this::users);
+	}
+
+	private String index(Request request, Response response) {
+		Map<String, Object> model = new HashMap<>();
+		return render(model, "index.html");
+	}
+
+	private String init(Request request, Response response) {
+		return render(new HashMap<>(), "userNames.html");
+	}
+
+	private String createNewGame(Request request, Response response) {
+		HashMap<String, Object> model = new HashMap<>();
+		model.put("id", request.params(":id"));
+		return render(model, "playingGame.html");
+	}
+
+	private String games(Request request, Response response) throws SQLException {
+		return gson.toJson(gamesDao.everyGames());
+	}
+
+	private String turn(Request request, Response response) throws SQLException {
+		int id = Integer.parseInt(request.params(":id"));
+		ChessGame game = ChessGame.createGameByMoves(moveDao.findMovesByGameId(id));
+		TurnDto turnDto = new TurnDto(game.turn());
+
+		HashMap<String, Object> model = new HashMap<>();
+		model.put("turn", turnDto);
+		return gson.toJson(model);
+	}
+
+	private String board(Request request, Response response) throws SQLException {
+		int id = Integer.parseInt(request.params(":id"));
+		ChessGame game = ChessGame.createGameByMoves(moveDao.findMovesByGameId(id));
+		UnitsDto units = new UnitsDto(game.board().getBoard());
+		return gson.toJson(units);
+	}
+
+	private String score(Request request, Response response) throws SQLException {
+		int id = Integer.parseInt(request.params(":id"));
+		ChessGame game = ChessGame.createGameByMoves(moveDao.findMovesByGameId(id));
+		ScoreDto score = new ScoreDto(game.status());
+		HashMap<String, Object> model = new HashMap<>();
+		model.put("score", score);
+		return gson.toJson(model);
+	}
+
+	private String move(Request request, Response response) throws SQLException {
+		int id = Integer.parseInt(request.params(":id"));
+		ChessGame game = ChessGame.createGameByMoves(moveDao.findMovesByGameId(id));
+		Map<String, String> map = gson.fromJson(request.body(), Map.class);
```

Comment:
내부 메소드 보니까 아래 같은 방식으로도 가능 한 것 같아요! 
코드에 경고가 뜨면 다른 방법이 없는지 객체 내부 메소드를 한번 살펴보세요! 

```suggestion
		Map<String, String> map = gson.fromJson(request.body(), new TypeToken<Map<String, String>>(){}.getType());
```

Reply:
처음 보는 부분인데 저걸 사용하니까 불필요한 형변환도 사라지고 명시적으로 의도를 드러낼 수 있는 것 같아서 좋은 것 같습니다. 감사합니다!!